subprojects/common/http-client/src/test/kotlin/com/avito/http/HttpClientProviderMetricsTest.kt

```diff
@@ -0,0 +1,130 @@
+package com.avito.http
+
+import com.avito.android.stats.SeriesName
+import com.avito.android.stats.StatsMetric
+import com.avito.android.stats.StubStatsdSender
+import com.avito.android.stats.TimeMetric
+import com.avito.http.internal.RequestMetadata
+import com.avito.logger.StubLoggerFactory
+import com.avito.time.StubTimeProvider
+import com.google.common.truth.Correspondence
+import com.google.common.truth.Truth.assertThat
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.net.SocketTimeoutException
+import java.util.concurrent.TimeUnit
+
+internal class HttpClientProviderMetricsTest {
+
+    private val loggerFactory = StubLoggerFactory
+
+    private val timeProvider = StubTimeProvider()
+
+    private val statsDSender = StubStatsdSender()
+
+    private val mockWebServer = MockWebServer()
+
+    private val metricNamesCorrespondence: Correspondence<StatsMetric, StatsMetric> = Correspondence.from(
+        { actual, expected ->
+            if (actual != null && expected != null) {
+                actual.name == expected.name
+            } else false
+        },
+        "metric names are equal"
+    )
+
+    private val doesNotMatter = 111L
+
+    @Test
+    fun `timeout metric`() {
```

Comment:
Let's test a few parallel different request to see that Event listener handle that

Reply:
will do